### PR TITLE
Fix variables in selectors when calling toString()

### PR DESCRIPTION
**Pull request by kaicataldo**
Also replacing nbd/util/extend with Object.assign while I'm at it :)

### CODE REVIEW

**Comment by hzoo** [comment]
👍

**Comment by nemtsov** [comment]
👍

File: src/Engine.js [hunk 1]
```diff
@@ -70,12 +69,17 @@ Engine = Class.extend({
...
 
-    this._state = payload;
+    this._state = Object.assign({}, payload, { focssVariables: this.variables });
     this._regenerateArrayRules();
 
     for (i = 0; i < this.rules.length; i++) {
       this.rules[i].process(this._getStateWithToggles(), this.extensions);
-      result += css.toString(this.rules[i].selector, this.rules[i].result);
+
+      const selector = this.rules[i].isComputed ?
```
Comment (mikesherov) [inline src/Engine.js]: FWIW, isn't there a function on Rule that returns the computed selector if the rule is computed? Use that.

File: test/behavior/toString.js [hunk 2]
```diff
@@ -44,5 +44,19 @@ describe('toString', function() {
...
+
+    it('when inserted rule uses variables', function() {
+      this._fox.insertVars({
+        class: 'foo',
+        bar: 'red'
+      });
+
+      this._fox.insert('.${focssVariables.class}', {
+        'max-width': 'width',
+        color: 'focssVariables.bar'
```
Comment (mikesherov) [inline test/behavior/toString.js]: `focssVariables.` seems overly verbose. Might be better to have e.g. `__var`

Comment (kaicataldo) [inline test/behavior/toString.js]: Totally agree - really like that idea!

File: src/Engine.js (hunk 1 above)
Comment (kaicataldo) [inline src/Engine.js]: I'll check tomorrow - if there is, I forgot about it. Good call.

**Pull request merged**